### DISCUSSION TITLE
How to add a button e.g  "Add New Customer" on razor page to show Modal Form with all DataGrid columns but other than header location?

### DISCUSSION THREAD

**Discussion by shakeeluetian** [Megabit/Blazorise#2739] [General]
I have a button "Add New Customer" on a razor page. i want to show a Modal on button click with all GridView Columns. I knew it is possible with `<NewCommandTemplate>` but it show button  in DataGrid header where i dont want to place a button for new row entery

**Comment by David-Moreira** [comment]
Hello @shakeeluetian 
Let me see if I understood that right. You'd like a "Add New Customer" button on your Datagrid, but not on the header row right?

Check out our demo for the Datagrid ButtonRow example:
https://bootstrapdemo.blazorise.com/tests/datagrid
![DgButtonRow](https://user-images.githubusercontent.com/22283161/127237842-acb9e95a-2407-4ae8-84b5-858486968a02.png)


Also here's the docs with a snippet:
https://blazorise.com/docs/extensions/datagrid/#datagrid-button-row
```
<ButtonRowTemplate>
    <Button Color="Color.Success" Clicked="@context.NewCommand.Clicked">New</Button>
    <Button Color="Color.Primary" Disabled="@(selectedEmployee is null)" Clicked="context.EditCommand.Clicked">Edit</Button>
    <Button Color="Color.Danger" Disabled="@(selectedEmployee is null)" Clicked="context.DeleteCommand.Clicked">Delete</Button>
    <Button Color="Color.Link" Clicked="@context.ClearFilterCommand.Clicked">Clear Filter</Button>
</ButtonRowTemplate>
```

Also nowadays you have access to the Datagrid's internal APIs, so you can even make your own button outside of the Datagrid's and trigger the New Operation. See #2503

**Reply by shakeeluetian** [reply]
I have
`<DataGrid TItem="CompanyModel" Data="@companies" ShowPager="true" Editable="true" CommandMode="DataGridCommandMode.ButtonRow" EditMode="DataGridEditMode.Popup" Striped="true" FixedHeader="true" CustomFilter="@OnCustomFilter" Style="font-size: 0.85rem;padding:0" Bordered="true" Responsive="true" Narrow="true" PageSize="10">`


Unfortunately `CommandMode="DataGridCommandMode.ButtonRow"` is not working

**Reply by David-Moreira** [reply]
Hello.
You also need to provide a ButtonRowTemplate just like in the snippet in my post above.

**Reply by stsrki** [reply]
Also, don't forget that with 0.9.4 we exposed most of the grid APIs so you can now use almost any internal command. Like `New()`, `Edit()`, `Save()`, etc. That means you can build your UI and place the button wherever you want.

**Reply by shakeeluetian** [reply]
Thank a lot for prompt response. Problem Resolved. Issue was in my code implementation.

**Reply by shakeeluetian** [reply]
@stsrki i have also done that. regards
